guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional/index.md

```diff
@@ -105,65 +104,31 @@ In the case that only one argument was passed, do not worry about how to prompt
 
 #### Relevant Links
 
-*   <a href='https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof' target='_blank' rel='nofollow'>typeof</a>
-*   <a href='https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments' target='_blank' rel='nofollow'>arguments object</a>
+*   [Typeof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof)
+*   [Arguments Object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments)
 
 ## ![:sunflower:](https://forum.freecodecamp.com/images/emoji/emoji_one/sunflower.png?v=3 ":sunflower:") Intermediate Code Solution:
 ```javascript
-    function addTogether() {
-      var args = new Array(arguments.length);
-      //Storing the arguments in an array
-      for(var i = 0; i < args.length; ++i) {
-          args[i] = arguments[i];
-        }
-     //Check for the arguments length
-     if(args.length == 2){
-        //If there are two arguments,check for the type of both arguments
-        //Use typeof to check the type of the argument(both should be numbers)
-        if(typeof args[0] !== 'number' || typeof args[1] !=='number' ){
-          return undefined;
-          }
-        return args[0]+args[1];
-       }
-     //When only one argument is provided
-     if(args.length == 1){
-         a= args[0];
-         //Check the  argument using typeof 
-        if(typeof a!=='number'){
-            return undefined;
-          }
-        else{
-           //Making use of closures 
-           return function(b){
-           //Checking the second argument 
-             if(typeof b !=='number'){
-               return undefined;
-               }
-             else
-               return a+b;
-              };
-          }
-        }
-    }
-
-    // test here
-    addTogether(2,3);
+function addTogether(first, second) {
+  if (typeof first !== 'number') {
+    return undefined;
+  }
+  const sum = second => (typeof second === 'number' ? first + second : undefined);
+  return typeof second === 'undefined' ? second => sum(second) : sum(second);
+}
+// test here
```

Comment:
What is supposed to happen as per the example when the typeof **second** is **null**(_typeof null === 'object'_)?

Reply:
Hi @mwangi-njuguna , it returns `undefined` as required `If either argument isn't a valid number, return undefined.` in the task.